src/spec.c

```diff
@@ -1062,6 +1115,11 @@ static int IndexSpec_AddFieldsInternal(IndexSpec *sp, StrongRef spec_ref, ArgsCu
     }
 
     FieldSpec *fs = IndexSpec_CreateField(sp, fieldName, fieldPath);
+    if (!fs) {
+      QueryError_SetErrorFmt(status, QUERY_ELIMIT, "Schema is currently limited to %d fields",
+                             sp->numFields);
+      goto reset;
+    }
```

Comment:
Breaking?
What would happen before in this case?

Reply:
There are two possible failure reasons:
1. We are going to overflow on the number of fields - in the past it would mean num of fields would become 0 which seems like stuff would stop working
2. rm_realloc failed due to lack of memory - in the past we would probably would have crashed when trying to access the field spec.

Reply:
Are you sure about (1)?
(2) makes sense and is the case that I'm less interested in here

Reply:
```cpp
fs->index = sp->numFields++;
```
Yes, pretty sure
Also note index is uint16_t and numFields is uint32_t.